packages/headless/src/controllers/insight/did-you-mean/headless-insight-did-you-mean.ts

```diff
@@ -10,20 +10,33 @@ import {
   buildCoreDidYouMean,
   DidYouMean,
   DidYouMeanState,
+  DidYouMeanProps,
+  DidYouMeanOptions,
 } from '../../core/did-you-mean/headless-core-did-you-mean.js';
 
-export type {QueryCorrection, WordCorrection, DidYouMean, DidYouMeanState};
+export type {
+  QueryCorrection,
+  WordCorrection,
+  DidYouMean,
+  DidYouMeanState,
+  DidYouMeanProps,
+  DidYouMeanOptions,
+};
 
 /**
  * The insight DidYouMean controller is responsible for handling query corrections.
  * When a query returns no result but finds a possible query correction, the controller either suggests the correction or
  * automatically triggers a new query with the suggested term.
  *
  * @param engine - The insight engine.
+ * @param props - The configurable `DidYouMean` properties.
  */
-export function buildDidYouMean(engine: InsightEngine): DidYouMean {
+export function buildDidYouMean(
+  engine: InsightEngine,
+  props: DidYouMeanProps = {}
```

Comment:
To avoid breaking changes, I would set the default value of props to `{queryCorrectionMode: 'legacy'}` this client will not be switched from `legacy` to `next` automatically without their intervention.

Reply:
I'm not sure if we should set default value is 'legacy', if we enforce client to use new mode, then using 'next' is better. And we can document for user that if you upgrade, you need to update yours. Otherwise more and more user using 'legacy', it's not easy for us in the future to remove it